packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -652,7 +669,7 @@ export class AppSyncModelVisitor<
       model.fields.forEach(field => {
         const connectionInfo = processConnectionsV2(field, model, this.modelMap);
         if (connectionInfo) {
-          if (connectionInfo.kind === CodeGenConnectionType.HAS_MANY || connectionInfo.kind === CodeGenConnectionType.HAS_ONE) {
+          if (connectionInfo.kind === CodeGenConnectionType.HAS_MANY) {
```

Comment:
The `HAS_ONE` connection no longer requires the connected field? Can you explain why? (possibly also add a unit test)

Reply:
removed the connected field from transformer V2 to match the behavior for the same schema in V1 (codegen seems to consider @connection for the equivalent V1 schema as a belongsTo connection type)
added a unit test for the change